### PR TITLE
Message size

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will fix a message size limitation in the ZMQ core.   the fix should be backwards compatible though older versions won't have the message get through completely.  The previous limitation was around 16 MB.  the tests now run to 1.5 GB.   There are other limitations for other core types that are not resolved by this PR.

### CODE REVIEW

**Comment by phlptp** [comment]
Fixes Issue #2698

File: src/helics/core/ActionMessage.cpp [hunk 1]
```diff
@@ -551,6 +572,10 @@ std::size_t ActionMessage::fromByteArray(const std::byte* data, std::size_t buff
             Tso.setBaseTimeCode(timecode);
         }
     }
+    if (size == 0x00FFFFFFUL) {
+        payload.append(stringData.back());
+        stringData.pop_back();
+    }
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:578]: Check for no string data.

File: src/helics/core/ActionMessage.cpp [hunk 2]
```diff
@@ -273,12 +290,16 @@ int ActionMessage::serializedByteCount() const
...
-    //   if (!stringData.empty()) {
+    if (!stringData.empty() || payload.size() > 0x00FFFFFF) {
+        size += 1;
+    }
     for (const auto& str : stringData) {
         // 4(to store the length)+length of the string
         size += static_cast<int>(sizeof(uint32_t) + str.size());
     }
-    // }
+    if (payload.size() > 0x00FFFFFF) {
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:300]:
```suggestion
    if (payload.size() >= 0x00FFFFFF) {
```

File: src/helics/core/ActionMessage.cpp [hunk 3]
```diff
@@ -273,12 +290,16 @@ int ActionMessage::serializedByteCount() const
     }
     size += static_cast<int>(payload.size());
     // add additional string data
-    //   if (!stringData.empty()) {
+    if (!stringData.empty() || payload.size() > 0x00FFFFFF) {
+        size += 1;
+    }
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:295]:
```suggestion
    size += 1;
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**